src/main/java/com/faforever/server/geoip/GeoIpService.java

```diff
@@ -63,10 +64,30 @@ public void updateDatabaseFile() throws IOException {
   }
 
   public Optional<String> lookupCountryCode(InetAddress inetAddress) {
+    Optional<CityResponse> city = lookupCity(inetAddress);
+    return city.isPresent() ? Optional.of(city.get().getCountry().getIsoCode()) : Optional.empty();
+  }
+
+  /**
+   * @return Time zone code as specified by the IANA
+   */
+  public Optional<String> lookupTimezone(InetAddress inetAddress) {
+    Optional<CityResponse> city = lookupCity(inetAddress);
+
+    if (!city.isPresent()) {
+      return Optional.empty();
+    } else {
+      return noCatch(() -> {
+        return Optional.ofNullable(city.get().getLocation().getTimeZone());
+      });
+    }
```

Comment:
IntelliJ already gives you some hints on what you can improve:
![image](https://cloud.githubusercontent.com/assets/1864183/24667413/aff768ee-1963-11e7-862b-174a71dfd42a.png)

However, the shortest solution is:
```
  public Optional<TimeZone> lookupTimezone(InetAddress inetAddress) {
    return lookupCity(inetAddress)
      .map(cityResponse -> cityResponse.getLocation().getTimeZone())
      .map(TimeZone::getTimeZone);
  }
```
noCatch isn't required here since nothing throws a checked exception.